soroban-react-dapp/contracts/authorized_greeting/src/lib.rs

```diff
@@ -0,0 +1,50 @@
+#![no_std]
+use soroban_sdk::{contract, contractimpl, Env, Symbol, symbol_short, String, Address, Map};
+
+const GREET: Symbol = symbol_short!("GREET");
+const AUTH_GRTRS: Symbol = symbol_short!("AUTH_GTRS");
+const ADMIN: Symbol = symbol_short!("ADMIN");
+
+#[contract]
+pub struct AuthorizedGreetingContract;
+
+#[contractimpl]
+impl AuthorizedGreetingContract {
+    pub fn set_admin(env: Env, admin: Address) {
+        env.storage().instance().set(&ADMIN, &admin);
+    }
+
+    pub fn get_admin(env: &Env) -> Address {
```

Comment:
Error could be better with a defined Defined Rust Error
https://developers.stellar.org/docs/build/smart-contracts/example-contracts/errors

Reply:
Added it right now!